main.go

```diff
@@ -63,7 +65,28 @@ func main() {
 		os.Setenv("OS_CLIENT_CONFIG_FILE", *osClientConfig)
 	}
 
-	http.HandleFunc("/", func(w http.ResponseWriter, r *http.Request) {
+	enabledExporters := 0
+	for service, disabled := range services {
+		if !*disabled {
+			_, err := exporters.EnableExporter(service, *prefix, *cloud, *disabledMetrics, *endpointType, *collectTime, *disableSlowMetrics, nil)
+			if err != nil {
+				// Log error and continue with enabling other exporters
+				log.Errorf("enabling exporter for service %s failed: %s", service, err)
+				continue
+			}
+			log.Infof("Enabled exporter for service: %s", service)
+			enabledExporters++
+		}
+	}
+
+	if enabledExporters == 0 {
+		log.Errorln("No exporter has been enabled, exiting")
+		os.Exit(-1)
+	}
+
+	sm := http.NewServeMux()
+	sm.Handle(*metrics, promhttp.Handler())
+	sm.HandleFunc("/", func(w http.ResponseWriter, r *http.Request) {
```

Comment:
why is this change required? lines 68 to 89

Reply:
You are right, i didn't dig enough to http lib.
Now it should be better.